libmymem.hpp

```diff
@@ -19,36 +20,81 @@ enum BucketSize{  // Enum for indexing into the correct size
 };
 
 
-struct Slab {       
-
-	int totalObj;
-	int freeObj;
-	bool bitmap[4096];
-	int initialize(){
-		totalObj=0;
-		freeObj=0;
-	}
-//	Bucket * bucket;
-
-	Slab * nextSlab;
-
-};
-
-
 struct Object {
-	Slab * parentSlab;
-	void * memory;
-	
+	void* parentSlab = NULL;
+	void * memory = NULL;
+	Object * nextpointer;
 };
 
+struct Slab {
+	int totalObj = 0 ; //4 bytes
+	int freeObj = 0; //4 bytes
+	int BucketSize = 0; //4 bytes
+	bool bitmap[4096]; //4096 bytes
+	void * bucket = NULL;
+	Slab * nextpointer;
+};
 struct Bucket {
 	size_t bucketSize;
 	Slab * slab;
 };
+void intialise_slab(Slab * s , int bs){
+	s->bucketSize = bs;
+	int used_space = 10;
```

Comment:
need to find space used by bucket size , another pointers ..and give a address to first Object pointer